File: 1198-find-smallest-common-element-in-all-rows/1198-find-smallest-common-element-in-all-rows.js
/**
 * @param {number[][]} mat
 * @return {number}
 */
var smallestCommonElement = function(mat) {
    
    let set = new Set();
    for(let x of mat[0]) {
        if(set.has(x)) continue;
        set.add(x);
    }
    
    for(let i = 1; i < mat.length; i++) {
        let prev = new Set();
        for(let x of mat[i]) {
            if(set.has(x)) {
                prev.add(x);
            }
        }
        set = prev;
    }
    
    return set.size > 0 ? Math.min(...set) : -1;
};

File: 1198-find-smallest-common-element-in-all-rows/1198-find-smallest-common-element-in-all-rows.java
class Solution {
    public int smallestCommonElement(int[][] mat) {
        final int m = mat.length;
        final int n = mat[0].length;
        HashMap<Integer, Integer> hmap = new HashMap<Integer, Integer>();
        for(int i = 0; i < m ; i++) {
            for(int j = 0; j < n; j++) {
               hmap.put(mat[i][j], hmap.getOrDefault(mat[i][j] , 0) + 1);
            }
        }
        int min = Integer.MAX_VALUE;
        for(int key : hmap.keySet()) {
            if(hmap.get(key) >= m) {
                min = Math.min(min, key);
            }
        }
        return min == Integer.MAX_VALUE ? -1 : min;
        
    }
}